### ISSUE TITLE
Replace mass shifts in peptide proforma strings during FragPipe and Sage parsing with modification names

### ISSUE THREAD

**Issue by Cajac102** [Proteobench/ProteoBench#444]
Might be a bug, modifications are still reported as mass shifts in FragPipe (e.g. "AC[[57.0215]]FHC[[57.0215]]ETC[[57.0215]]K|Z=3") and Sage (e.g. "AAAIGIDLGTTYSC[[+57.0215]]VGVFQHGK|Z=3
") result tables while in MaxQuant and AlphaPept, they are reported with names (e.g. "AAC[Carbamidomethyl]LPLPGYR|Z=2.0
"). The replacement dicts are already in the .toml.

**Comment by mlocardpaulet** [comment]
Yes, we decided not to homogenise the modifications at this stage. 
The thing is: every software and version can have its own way of encoding modifications. And for ProteoBench it does not really matter since we do not compare between tool. 
This mean that if we want to compare outputs of different tools (or versions) (for example for the paper ;)), we need to homogenise the PTMs afterwards.

We can totally discuss changing this.

**Comment by Cajac102** [comment]
I see, that makes total sense. If we only look at the total number of peptidoforms it doesn't matter at all.
From the code it is intended to homogenise, but the regex was off a bit. I fixed it for myself since I want to do the UpSetplots for the paper, if we choose to change it we can just merge it :)

**Comment by mlocardpaulet** [comment]
It never hurts to improve the code ;)

**Comment by Cajac102** [comment]
Two more differences I ran into:
-For AlphaPept, the charges are floats. I added parsing them to integers to make the proforma strings consistent.
-For MaxQuant: Right now, fixed modifications are not included in the proforma string. This might be a bit more effort to include, so I only hacked it together locally for the plot.

**Comment by Cajac102** [comment]
As Robbin already predicted, simply replacing the mass shifts with the modification names does not always work due to differences in precision. I have a sage file now where Carbamidomethylation is "+57.021465" instead of just "+57.0215", leading to the replacing not working.

**Comment by Cajac102** [comment]
Fixed temporarily in commit 7ed16c4, but we might need something more sophisticated for the future
